src/CoreWCF.Primitives/tests/DispatcherClient/DispatcherTransportBindingElement.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.ServiceModel.Channels;
+
+namespace DispatcherClient
+{
+    internal class DispatcherTransportBindingElement<TService, TContract> : TransportBindingElement where TService : class
+    {
+        private Action<IServiceCollection> _configureServices;
+
+        internal DispatcherTransportBindingElement(Action<IServiceCollection> configureServices)
+        {
+            _configureServices = configureServices;
+        }
+
+        public override IChannelFactory<TChannel> BuildChannelFactory<TChannel>(BindingContext context)
+        {
+            return new DispatcherChannelFactory<TChannel, TService, TContract>(_configureServices);
+        }
+
+        public override bool CanBuildChannelFactory<TChannel>(BindingContext context)
+        {
+            if (typeof(TChannel) == typeof(IRequestChannel) ||
+                typeof(TChannel) == typeof(IRequestSessionChannel))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        public override string Scheme => "COREWCF";
```

Comment:
Shouldn't Scheme be "corewcf"(small letters), for consistency purposes?

Reply:
You're right. For some reason I had in my head that the scheme was uppercase in other transports but I was wrong.